### PR TITLE
 fix: make project path OS-independent

### PR DESCRIPTION
**Pull request by i4ki**
Fixes tests in packages `hcl`, `hcl/eval`, `generate` and `stack`.

### CODE REVIEW

File: cmd/terramate/e2etests/runner_unix_test.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import (
+	"os"
+	"syscall"
+
+	"github.com/madlambda/spells/assert"
+)
+
+// build flags for unix below were taken from:
+// https://github.com/golang/go/blob/master/src/cmd/dist/build.go#L943
+// We can replace them by "unix" if we support only go >= 1.19 later.
```
Comment (i4ki) [inline cmd/terramate/e2etests/runner_unix_test.go:28]: @katcipis oh gosh, the gofmt moved the `// go:build` line to the top and now the comment is orphaned haha I will fix.

File: hcl/fmt_test.go [hunk 2]
```diff
@@ -1458,12 +1455,21 @@ func TestFormatTreeFailsOnNonAccessibleFile(t *testing.T) {
 		b = 3
 	}`)
 
-	assert.NoError(t, os.Chmod(filepath.Join(tmpdir, filename), 0))
+	test.Chmod(t, filepath.Join(tmpdir, filename), 0)
+	defer func() {
+		test.Chmod(t, filepath.Join(tmpdir, filename), 0755)
+	}()
```
Comment (katcipis) [inline hcl/fmt_test.go:1461]:
```suggestion
	defer test.Chmod(t, filepath.Join(tmpdir, filename), 0755)
```

File: hcl/fmt_test.go [hunk 3]
```diff
@@ -1432,18 +1432,15 @@ func TestFormatTreeReturnsEmptyResultsForEmptyDir(t *testing.T) {
...
-
 func TestFormatTreeFailsOnNonAccessibleSubdir(t *testing.T) {
 	const subdir = "subdir"
 	tmpdir := t.TempDir()
 	test.Mkdir(t, tmpdir, subdir)
 
 	assert.NoError(t, os.Chmod(filepath.Join(tmpdir, subdir), 0))
+	defer func() {
+		assert.NoError(t, os.Chmod(filepath.Join(tmpdir, subdir), 0755))
+	}()
```
Comment (katcipis) [inline hcl/fmt_test.go:1443]:
```suggestion
	defer test.Chmod(t, filepath.Join(tmpdir, subdir), 0755)
```

File: hcl/fmt_test.go [hunk 4]
```diff
@@ -1432,18 +1432,15 @@ func TestFormatTreeReturnsEmptyResultsForEmptyDir(t *testing.T) {
...
-	_, err := hcl.FormatTree(filepath.Join(tmpdir, "non-existent"))
-	assert.Error(t, err)
-}
-
 func TestFormatTreeFailsOnNonAccessibleSubdir(t *testing.T) {
 	const subdir = "subdir"
 	tmpdir := t.TempDir()
 	test.Mkdir(t, tmpdir, subdir)
 
 	assert.NoError(t, os.Chmod(filepath.Join(tmpdir, subdir), 0))
```
Comment (katcipis) [inline hcl/fmt_test.go:1440]:
```suggestion
	test.Chmod(t, filepath.Join(tmpdir, subdir), 0)
```

**Review by katcipis — COMMENTED** [review]
few comments, still reviewing :rocket:

File: project/project.go [hunk 5]
```diff
@@ -15,21 +15,64 @@
...
+		vals = append(vals, p.String())
+	}
+	return vals
+}
+
 // Metadata represents project wide metadata.
 type Metadata struct {
 	rootdir string
-	stacks  []string
+	stacks  Paths
```
Comment (katcipis) [inline project/project.go:71]: Noice :rocket:

File: stack/stack.go [hunk 6]
```diff
@@ -366,6 +366,15 @@ func Reverse(stacks List) {
 	}
 }
 
+// Paths returns the project paths from the stack list.
+func (l List) Paths() project.Paths {
+	strs := make(project.Paths, len(l))
+	for i, s := range l {
+		strs[i] = s.Path()
+	}
+	return strs
```
Comment (katcipis) [inline stack/stack.go:375]:
```suggestion
	paths := make(project.Paths, len(l))
	for i, s := range l {
		paths[i] = s.Path()
	}
	return paths
```

**Review by katcipis — APPROVED** [review]
:rocket: :rocket: :rocket:

**Pull request merged**